bpf/lib/encrypt.h

```diff
@@ -16,56 +16,62 @@
 static __always_inline int
 do_decrypt(struct __ctx_buff *ctx, __u16 proto)
 {
+	void *data, *data_end;
+	__u8 protocol = 0;
 	bool decrypted;
-
-	decrypted = ((ctx->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT);
-	if (!decrypted) {
-		void *data, *data_end;
-		__u8 protocol = 0;
 #ifdef ENABLE_IPV6
-		struct ipv6hdr *ip6;
+	struct ipv6hdr *ip6;
 #endif
 #ifdef ENABLE_IPV4
-		struct iphdr *ip4;
+	struct iphdr *ip4;
 #endif
 
-		switch (proto) {
+	decrypted = ((ctx->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT);
+
+	switch (proto) {
 #ifdef ENABLE_IPV6
-		case bpf_htons(ETH_P_IPV6):
-			if (!revalidate_data_pull(ctx, &data, &data_end, &ip6))
-				break;
-			protocol = ip6->nexthdr;
-			break;
+	case bpf_htons(ETH_P_IPV6):
+		if (!revalidate_data_pull(ctx, &data, &data_end, &ip6))
+			return CTX_ACT_OK;
```

Comment:
Do we need to clear `ctx->mark` in these cases?

Reply:
Probably a nice touch to clear the mark so we don't try to decrypt a decrypted packet if the data_pull fails after encryption.